src/Umbraco.Web.UI.Client/src/views/propertyeditors/blockgrid/prevalue/umbBlockGridAreaEditor.component.js

```diff
@@ -45,46 +75,78 @@
-                
+            function onSortSync() {
+                $scope.$evalAsync();
+                setDirty();
             }
 
-            const gridContainerEl = $element[0].querySelector('.umb-block-grid-area-editor__grid-wrapper');
-
-            const sortable = Sortable.create(gridContainerEl, {
-                sort: true,  // sorting inside list
-                animation: 150,  // ms, animation speed moving items when sorting, `0` — without animation
-                easing: "cubic-bezier(1, 0, 0, 1)", // Easing for animation. Defaults to null. See https://easings.net/ for examples.
-                cancel: '',
-                draggable: ".umb-block-grid-area-editor__area",  // Specifies which items inside the element should be draggable
-                ghostClass: "umb-block-grid-area-editor__area-placeholder",
-                onAdd: function (evt) {
-                    _sync(evt);
-                    $scope.$evalAsync();
-                },
-                onUpdate: function (evt) {
-                    _sync(evt);
-                    $scope.$evalAsync();
-                }
-            });
+            function resolveVerticalDirection(data) {
+
+                /** We need some data about the grid to figure out if there is room to be placed next to the found element */
+                const approvedContainerComputedStyles = getComputedStyle(data.containerElement);
+                const gridColumnGap = Number(approvedContainerComputedStyles.columnGap.split("px")[0]) || 0;
+                const gridColumnNumber = vm.rootLayoutColumns;
+    
+                const foundElColumns = parseInt(data.relatedElement.dataset.colSpan, 10);
+                const currentElementColumns = data.item.columnSpan;
 
-            // TODO: setDirty if sort has happend.
+                if(currentElementColumns >= gridColumnNumber) {
+                    return true;
+                }
+    
+                // Get grid template:
+                const approvedContainerGridColumns = approvedContainerComputedStyles.gridTemplateColumns.trim().split("px").map(x => Number(x)).filter(n => n > 0).map((n, i, list) => list.length === i ? n : n + gridColumnGap);
+    
+                // ensure all columns are there.
+                // This will also ensure handling non-css-grid mode,
+                // use container width divided by amount of columns( or the item width divided by its amount of columnSpan)
+                let amountOfColumnsInWeightMap = approvedContainerGridColumns.length;
+                const amountOfUnknownColumns = gridColumnNumber-amountOfColumnsInWeightMap;
+                if(amountOfUnknownColumns > 0) {
+                    let accumulatedValue = getAccumulatedValueOfIndex(amountOfColumnsInWeightMap, approvedContainerGridColumns) || 0;
+                    const layoutWidth = data.containerRect.width;
+                    const missingColumnWidth = (layoutWidth-accumulatedValue)/amountOfUnknownColumns;
```

Comment:
Consider if we should check if `missingColumnWidth` is less than or equal to zero before going in to the while loop.

Reply:
That's properly a good one, I've added it now